icu4c/source/common/unicode/rbbi.h

```diff
@@ -235,7 +240,7 @@ class U_COMMON_API RuleBasedBreakIterator /*U_FINAL*/ : public BreakIterator {
      * @see #getBinaryRules
      * @stable ICU 2.8
      */
-    RuleBasedBreakIterator(UDataMemory* image, UErrorCode &status);
+    RuleBasedBreakIterator(UDataMemory* image, UErrorCode &status, const char* type);
```

Comment:
This is not an acceptable change. Adding a new parameter will break pre-existing client. You could propose to add an new constructor which take one more arg but we need to keep the current one around to maintain backward compatability.

Reply:
Fix by adding a private constructor.